x/evm/keeper/params.go

```diff
@@ -2,57 +2,114 @@ package keeper
+}
+
+// SetParams Sets the EVM params each in its individual key for better get performance
+func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
+	if err := params.Validate(); err != nil {
+		return err
 	}
+	store := ctx.KVStore(k.storeKey)
+
+	chainCfgBz := k.cdc.MustMarshal(&params.ChainConfig)
+	extraEIPsBz := k.cdc.MustMarshal(&params.ExtraEips)
+	evmDenomBz := k.cdc.MustMarshal(&gogotypes.StringValue{Value: params.EvmDenom})
+	allowUnprotectedTxsBz := k.cdc.MustMarshal(&gogotypes.BoolValue{Value: params.AllowUnprotectedTxs})
+	enableCallBz := k.cdc.MustMarshal(&gogotypes.BoolValue{Value: params.EnableCall})
+	enableCreateBz := k.cdc.MustMarshal(&gogotypes.BoolValue{Value: params.EnableCreate})
+
+	store.Set(types.ParamStoreKeyExtraEIPs, extraEIPsBz)
+	store.Set(types.ParamStoreKeyChainConfig, chainCfgBz)
+	store.Set(types.ParamStoreKeyEVMDenom, evmDenomBz)
+	store.Set(types.ParamStoreKeyAllowUnprotectedTxs, allowUnprotectedTxsBz)
+	store.Set(types.ParamStoreKeyEnableCall, enableCallBz)
+	store.Set(types.ParamStoreKeyEnableCreate, enableCreateBz)
 
-	return params
+	return nil
 }
 
-// SetParams sets the evm parameters to the param space.
-func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
-	k.paramSpace.SetParamSet(ctx, &params)
+// GetExtraEIPs returns the extra EIPs enabled on the chain.
+func (k Keeper) GetExtraEIPs(ctx sdk.Context) types.ExtraEIPs {
+	var extraEIPs types.ExtraEIPs
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ParamStoreKeyExtraEIPs)
+	if bz == nil {
+		return extraEIPs
+	}
+	k.cdc.MustUnmarshal(bz, &extraEIPs)
+	return extraEIPs
 }
 
 // GetChainConfig returns the chain configuration parameter.
 func (k Keeper) GetChainConfig(ctx sdk.Context) types.ChainConfig {
 	var chainCfg types.ChainConfig
-	k.paramSpace.GetIfExists(ctx, types.ParamStoreKeyChainConfig, &chainCfg)
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ParamStoreKeyChainConfig)
+	if bz == nil {
+		return chainCfg
+	}
+	k.cdc.MustUnmarshal(bz, &chainCfg)
 	return chainCfg
 }
 
 // GetEVMDenom returns the EVM denom.
 func (k Keeper) GetEVMDenom(ctx sdk.Context) string {
-	var evmDenom string
-	k.paramSpace.GetIfExists(ctx, types.ParamStoreKeyEVMDenom, &evmDenom)
-	return evmDenom
+	var evmDenom gogotypes.StringValue
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ParamStoreKeyEVMDenom)
+	if bz == nil {
+		return evmDenom.Value
+	}
+	k.cdc.MustUnmarshal(bz, &evmDenom)
+	return evmDenom.Value
 }
 
 // GetEnableCall returns true if the EVM Call operation is enabled.
 func (k Keeper) GetEnableCall(ctx sdk.Context) bool {
-	var enableCall bool
-	k.paramSpace.GetIfExists(ctx, types.ParamStoreKeyEnableCall, &enableCall)
-	return enableCall
+	var enableCall gogotypes.BoolValue
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ParamStoreKeyEnableCall)
+	if bz == nil {
+		return enableCall.Value
+	}
+	k.cdc.MustUnmarshal(bz, &enableCall)
+	return enableCall.Value
```

Comment:
see comment above. You can use `store.Has` which is cheaper (performance and gas-wise)